File: 002/002.cpp
#include <iostream>

using namespace std;

// n! = 1 * 2 * ... * (n - 1) * n
int factorial1(int number) {
    int result = 1;

    for (int i = 1; i <= number; i++)
        result *= i;

    return result;
}

// n! = 1 * 2 * ... * (n - 2) * (n-1)! * n
int factorial2(int number) {
    if (number <= 1)
        return 1;

    return factorial2(number - 1) * number;
}

int main() {
    cout << "Enter a number: ";

    string input;
    cin >> input;

    int number = stoi(input);

    cout << input << "! = " << factorial1(number) << endl;
    cout << input << "! = " << factorial2(number) << endl;

    return 0;
}

File: 002/002.go
package main

import "fmt"

func factorial1(number int) int {
	var result int = 1

	for num := 1; num <= number; num++ {
		result *= num
	}

	return result
}

func factorial2(number int) int {
	if number <= 1 {
		return 1
	}

	return factorial2(number-1) * number
}

func main() {
	fmt.Print("Enter a number: ")

	var number int = 0
	fmt.Scanf("%d", &number)

	factorial := factorial1(number)
	fmt.Printf("%d! = %d\n", number, factorial)

	factorial = factorial2(number)
	fmt.Printf("%d! = %d\n", number, factorial)
}
